public/app/core/utils/richHistory.ts

```diff
@@ -251,3 +259,31 @@ export function mapQueriesToHeadings(query: RichHistoryQuery[], sortOrder: SortO
 
   return mappedQueriesToHeadings;
 }
+
+/* Create datasource list with images. If specific datasource retrieved from Rich history is not part of
+ * exploreDatasources add generic datasource image and add property isRemoved = true.
+ */
+export function createDatasourceListWithImages(queriesDatasources: string[]) {
```

Comment:
Just to double check, we already should have something like this for datasource picker in explore (and panel edit) where we show both name and icon so I wonder if we do not have basically the same function somewhere already.

Reply:
Yes, you are right. This is slightly different functionality cause is creates structured data source list with required properties. I have updated the naming, so it makes more sense.